src/testdir/test_messages.vim

```diff
@@ -268,3 +268,7 @@ func Test_message_more()
   call term_sendkeys(buf, ':q!')
   call StopVimInTerminal(buf)
 endfunc
+
+func Test_null_list()
+  echom test_null_list()
```

Comment:
How about calling function Test_null() and test not only test_null_list() but also test_null_dict() (which also crashed before your fix, I did not check whether your change fixes it), test_null_blob(), test_null_job(), test_null_string(), test_null_channel(), test_null_partial().